lib/console.js

```diff
@@ -49,9 +55,32 @@ Console.prototype.error = Console.prototype.warn;
 
 
 Console.prototype.dir = function(object, options) {
-  this._stdout.write(util.inspect(object, util._extend({
-    customInspect: false
-  }, options)) + '\n');
+  this._stdout.write((this._group ? util.inspect(object, util._extend({
+        customInspect: false }, options)).replace(/^/gm, this._prefix) :
+          util.inspect(object, util._extend({ customInspect: false },
+              options)).replace(/^/gm, this._prefix)) + '\n');
+};
+
+
+Console.prototype.group = function() {
+  if (arguments.length) {
+    this.log.apply(this, arguments);
+  }
+  this._group++;
+  this._prefix = '  '.repeat(this._group);
+};
+
+
+Console.prototype.groupCollapsed = function() {
+  this.group.apply(this, arguments);
+};
+
+
+Console.prototype.groupEnd = function() {
+  if ( this._group > 0 ) {
+    this._group--;
+    this._prefix = (this._group ? '  '.repeat(this._group) : '');
```

Comment:
@3y3 So to summarize:
* You want to get rid of the `( a ? b : c )` notation and replace with `if else` structures
* Use `+= '  '` concatenation in `group()` to modify `_prefix` string
* Use `String.splice` in `groupEnd()` to modify `_prefix` string

You have a valid approach, but here I did not choose that. Lets give the maintainers some time to comment on this and give us direction :)